forge-game/src/main/java/forge/game/ability/effects/ChangeZoneEffect.java

```diff
@@ -663,9 +663,26 @@ private void changeKnownOriginResolve(final SpellAbility sa) {
                 }
                 if (sa.isKeyword(Keyword.UNEARTH) && movedCard.isInPlay()) {
                     movedCard.setUnearthed(true);
-                    movedCard.addChangedCardKeywords(Lists.newArrayList("Haste"), null, false, game.getNextTimestamp(), null);
+
+                    final Card eff = createEffect(sa, sa.getActivatingPlayer(), "Unearth Effect", hostCard.getImageKey());
+
+                    // It gains haste.
+                    String s = "Mode$ Continuous | Affected$ Card.IsRemembered | EffectZone$ Command | AddKeyword$ Haste";
+                    eff.addStaticAbility(s);
+
+                    // If it would leave the battlefield, exile it instead of putting it anywhere else.
+                    addLeaveBattlefieldReplacement(eff, "Exile");
+
+                    eff.addRemembered(movedCard);
+
+                    // Add forgot trigger
+                    addExileOnMovedTrigger(eff, "Battlefield");
```

Comment:
adding both is redundant

Reply:
What would you prefer? With only one defined Creature, the Command might be better?

Reply:
yea then no trigger needs to be parsed and tested